packages/store/addon/-private/record-arrays/identifier-array.ts

```diff
@@ -746,12 +746,12 @@ if (DEPRECATE_ARRAY_LIKE) {
   IdentifierArray.prototype.filterBy = function (key: string, value?: unknown) {
     deprecateArrayLike(this.DEPRECATED_CLASS_NAME, 'filterBy', 'filter');
     if (arguments.length === 2) {
-      return this.filter((value) => {
-        return Boolean(get(value, key));
+      return this.filter((record) => {
+        return get(record, key) === value;
       });
     }
-    return this.filter((value) => {
-      return Boolean(get(value, key));
+    return this.filter((record) => {
+      return Boolean(get(record, key));
```

Comment:
:facepalm: you may want to check the other implementations to make sure I didn't make a similar mistake with them

Reply:
😆 copy paste so useful, so easy to mess up. I eyeballed the rest and they looked good, decided not to wrote a bunch of soon to be deprecated tests though seemed like overkill.